console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersController.java

```diff
@@ -676,6 +686,16 @@ private Account createAccountFromRequestBody(ServletInputStream is) throws Illeg
 			}
 		}
 
+		String privacyPolicyAgreementDate = RequestUtil.getFieldValue(json, UserSchema.PRIVACY_POLICY_AGREEMENT_DATE_KEY);
```

Comment:
Parse errors should be discarded at validation time ?

Reply:
I don't understand, sorry. What is the proposal?

Reply:
Json should be first validated, I think it is, looking for example for mandatory field, is it ?
In this case we should also validate that date format are ok so to avoid later parse exception ?
One should also try to reuse same code in account controller for update and modify service (building an account from a json ?).

Just saying that small refactor steps are welcomed in my opinion.

Reply:
OK, I will work on it.

Reply:
This sais I am probably messing things with OrgControler as 

https://github.com/georchestra/georchestra/pull/2546/files/4fb587e287128730054495a5effaf504a9c9f842#diff-0ceccf060c18478dcaeeef338384f688L596

states for example that if org is messing for json, let it unchanged.

Reply:
> In this case we should also validate that date format are ok so to avoid later parse exception ?

I think it's what the exceptions are for. The two expected behaviors are:

- empty field ("") -> we do not set the privacyPolicyAgreementDate field in the Account object
- some value -> it should be formatted as expected ('2007-12-03') by LocalData.parse(), or an exception would be thrown because the JSON is malformed.

(note that I copied and adapted the code above, for shadowExpire)

Reply:
> One should also try to reuse same code in account controller for update and modify service (building an account from a json ?).

Yes, I agree.

Reply:
> an exception would be thrown because the JSON is malformed

I wonder if we are managing the errors correctly in that REST API (I don't think so). I imagine that in https://github.com/georchestra/georchestra/blob/master/console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersController.java#L324 and https://github.com/georchestra/georchestra/blob/master/console/src/main/java/org/georchestra/console/ws/backoffice/users/UsersController.java#L392 endpoints, we should manage the case where the JSON is malformed, and return the corresponding HTTP error code, instead of failing with an exception on the server side.